hack/verify-structured-logging.sh

```diff
@@ -46,9 +50,31 @@ done < <(echo "${all_packages[@]}" | tr " " "\n")
 
 ret=0
 echo -e "\nRunning structured logging static check on all packages"
-GOOS=linux    logcheck "${packages[@]}" || ret=$?
-GOOS=windows  logcheck "${packages[@]}" || ret=$?
+
+# Function to run logcheck and capture output
+run_logcheck() {
+  local os="$1"
+  shift
+  local output
+  output=$(GOOS="$os" logcheck "$@" 2>&1)
+  # TODO: Add more checks as needed
+  # Currently, we only check for unstructured logging
+  # more detail: https://github.com/kubernetes-sigs/scheduler-plugins/pull/831#discussion_r1835553079
+  if echo "$output" | grep -q 'unstructured logging function'; then
+    echo "Error: Unstructured logging detected. Please fix the issues."
```

Comment:
I think we need to use a variable in L63 to hold the errorous entries, so as to dump below L64, right? (pseudo code below, please adjust/evalaute)

```bash
errors=$(echo "$output" | grep -q 'unstructured logging function')
if [[ $? -ne 0 ]]; then
   echo "Error blabla"
   echo "$errors"
```